a fallback matcher that is guaranteed to come last

sbeam opened the pull request:
Since matchers can be added at any point at runtime, it's not possible to guarantee that a generic catch-all, eg `{{ (.*?) }}` will come last when a template is processed. If it doesn't come last, it probably will clobber the template for other matchers that should match.

sbeam committed:
add_fallback to provide a matcher/handler that always comes last

markbates commented:
Looks good. I'll merge this in tomorrow. 
—
Mark Bates

On Thu, Mar 7, 2013 at 8:08 PM, Sam Beam notifications@github.com wrote:

> Since matchers can be added at any point at runtime, it&#39;s not possible to guarantee that a generic catch-all, eg `{{ (.*?) }}` will come last when a template is processed. If it doesn&#39;t come last, it probably will clobber the template for other matchers that should match.
> You can merge this Pull Request by running:
>   git pull https://github.com/sbeam/braai master
> Or you can view, comment on it, or merge it online at:
>   https://github.com/markbates/braai/pull/7
> -- Commit Summary --
> - add_fallback to provide a matcher/handler that always comes last
>   -- File Changes --
>   M lib/braai/context.rb (40)
>   M lib/braai/handlers.rb (2)
>   M lib/braai/handlers/base.rb (2)
>   M lib/braai/handlers/iteration.rb (2)
>   M lib/braai/matchers.rb (6)
>   M lib/braai/template.rb (8)
>   M spec/braai/context_spec.rb (5)
>   M spec/braai/handlers/base_spec.rb (6)
>   M spec/braai/template_spec.rb (26)
>   -- Patch Links --
>   https://github.com/markbates/braai/pull/7.patch
>   https://github.com/markbates/braai/pull/7.diff

sbeam committed:
improve test to only work with add_callback

for some reason, adding the generic regex last makes it come first in
matchers. Now, if you change add_fallback to map, it would fail.

sbeam committed:
cleanup test and clear matchers and fallback to avoid crossover
